samples/Jenkinsfile

```diff
@@ -1,37 +1,172 @@
-node {
-    stage('Checkout') {
-        checkout scm
-    }
+// node {
+//     stage('Checkout') {
+//         checkout scm
+//     }
 
-    stage('Build') {
-        mvn 'clean install -DskipTests'
-    }
+//     stage('Build') {
+//         mvn 'clean install -DskipTests'
+//     }
 
-    stage('Unit Test') {
-        mvn 'test'
-    }
+//     stage('Unit Test') {
+//         mvn 'test'
+//     }
+
+//     stage('Integration Test') {
+//         mvn 'verify -DskipUnitTests -Parq-wildfly-swarm '
+//     }
+// }
+
+// def mvn(def args) {
+//     def mvnHome = tool 'M3'
+//     def javaHome = tool 'JDK8'
 
-    stage('Integration Test') {
-        mvn 'verify -DskipUnitTests -Parq-wildfly-swarm '
+//     // Apache Maven related side notes:
+//     // --batch-mode : recommended in CI to inform maven to not run in interactive mode (less logs)
+//     // -V : strongly recommended in CI, will display the JDK and Maven versions in use.
+//     //      Very useful to be quickly sure the selected versions were the ones you think.
+//     // -U : force maven to update snapshots each time (default : once an hour, makes no sense in CI).
+//     // -Dsurefire.useFile=false : useful in CI. Displays test errors in the logs directly (instead of
+//     //                            having to crawl the workspace files to see the cause).
+
+//     // Advice: don't define M2_HOME in general. Maven will autodetect its root fine.
+//     // See also
+//     // https://github.com/jenkinsci/pipeline-examples/blob/master/pipeline-examples/maven-and-jdk-specific-version/mavenAndJdkSpecificVersion.groovy
+//     withEnv(["JAVA_HOME=${javaHome}", "PATH+MAVEN=${mvnHome}/bin:${env.JAVA_HOME}/bin"]) {
+//         sh "${mvnHome}/bin/mvn ${args} --batch-mode -V -U -e -Dsurefire.useFile=false"
+//     }
+// }
+pipeline {
+    agent any
+    
+    tools {
+        ant 'AntNew'
+        gradle 'GradleNew'
+        maven 'M3'
     }
-}
+    
+    stages {
+        stage('Plugin Build') {
+            steps {
+                script {
+                    // Perform the checkout
+                    checkout([
+                        $class: 'GitSCM',
+                        branches: [[name: '*/main']],
+                        doGenerateSubmoduleConfigurations: false,
+                        extensions: [],
+                        submoduleCfg: [],
+                        userRemoteConfigs: [[url: 'https://github.com/Anshika2203/CombinedAllBuilds.git']]
```

Comment:
```suggestion
                        userRemoteConfigs: [[url: 'https://github.com/harness-community/jenkins-to-harness.git']]
```

Reply:
let's have the test files in a common repo over personal-github accounts. @Anshika2203 to push the files to https://github.com/harness-community/jenkins-to-harness repo.